index.bs

```diff
@@ -1965,6 +1965,38 @@ See also:
 
 * <a href="https://www.w3.org/2001/tag/doc/promises-guide">Writing Promise-Using Specifications</a>
 
+<h4 id="sync-callbacks">When to use callback functions instead</h4>
+
+<p>If an algorithm requires that callback functions respond
+synchronously (e.g. to take or veto some action), then that is a
+synchronous API, and should not be modelled with a promise. Instead,
+use an explicit callback or event.</p>
+
+<p>In Promise reaction callbacks, you may not require action within
+the same microtask.</p>
+
+<p>This protects promise composition: patterns like <code>await</code>,
+wrapper helpers, logging, <code>Promise.race()</code>, or multiple
+<code>.then()</code> branches all queue an extra microtask and must
+not cause action at a distance.</p>
+
+<div class=example>
+Calling <code>event.preventDefault()</code> in an event handler,
+or choosing a response object inside a
+<code>fetchEvent.respondWith()</code> callback.
```

Comment:
This should be a sentence. How about:

```suggestion
If developer-supplied code needs to be able to call <code>event.{{Event/preventDefault()}}</code>
or choose a response object with
<code>fetchEvent.{{FetchEvent/respondWith()}}</code>,
it needs to be a callback rather than a reaction passed to <code>Promise.{{Promise/then()}}</code>.
```